openmc/mesh.py

```diff
@@ -75,8 +75,8 @@ def __eq__(self, mesh2):
             return False
 
         # Check lower left / upper right
-        elif self._lower_left != mesh2.lower_left and \
-             self._upper_right != mesh2.upper_right:
+        elif (self._lower_left != mesh2.lower_left).all() and \
```

Comment:
Can you apply this same fix to the comparisons of the `width` and `dimension` attributes above?